frontend/client/__tests__/reset-password.test.js

```diff
@@ -0,0 +1,143 @@
+import { mount } from 'enzyme'
+import React from 'react'
+import { act } from 'react-dom/test-utils'
+import ChangePasswordModal from '../src/components/ChangePasswordModal'
+import ResetPasswordModal from '../src/components/ResetPasswordModal'
+import { createStore } from '../src/store'
+import { auth } from '../src/store/firebase'
+
+// Mock auth.currentUser.updatePassword()
+jest.mock('../src/store/firebase', () => {
+  return {
+    auth: {
+      currentUser: {
+        email: 'test@test.com',
+        updatePassword: jest.fn(() => Promise.resolve(true)),
+        reauthenticateWithCredential: jest.fn(() => Promise.resolve(true)),
+      },
+      EmailAuthProvider: { credential: () => {} },
+      setPersistence: () => {},
+      onAuthStateChanged: () => {},
+    },
+    // Nested function hell mocking the db
+    db: {
+      collection: () => {
+        return {
+          doc: () => {
+            return { update: () => Promise.resolve(true) }
+          },
+        }
+      },
+    },
+  }
+})
+
+describe('reset password', () => {
+  const waitForComponentToPaint = async (wrapper) => {
+    await act(async () => {
+      await new Promise((resolve) => setTimeout(resolve, 0))
+      wrapper.update()
+    })
+  }
+
+  test.each([
+    ['Reset Password Modal success', ResetPasswordModal, 'password1', 'password1'],
```

Comment:
When I change `'password 1'` to `'password 2'` and vice versa, the tests still pass. Based on my experience with using `jest` for the firebase tests, this typically means that there's some asynchronous call that occurs, and `jest` just "jumps over it", reaches the end of the function, and says the test passed

Reply:
I think the way this is setup using test.each, either combination of password 1 / password 1 or password 1 / password 2 will work. But I believe the functionality is being tested.